### ISSUE TITLE
Mac client doesnt update properly

### ISSUE THREAD

**Issue by calebrepkes** [a85/POSTMan-Chrome-Extension#2489]
1. Postman Version: 4.8.0
 2. App (Chrome app or Mac app): Mac App
 3. OS details: OS X 16.0.0 / x64   &&  macOS Sierra 10.12
 4. Is the Interceptor on and enabled in the app: Automatically yes. 
 5. Did you encounter this recently, or has this bug always been there: Had it too when trying to upgrade from 4.7.x to 4.8.0
 6. Expected behaviour: Update properly
 7. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app): Nothing to log
 8. Screenshots (if applicable)

 Open Mac App 
Wait for popup to show, there is an update available. 
Click update now 
Wait until update is downloaded
Click Restart and install update
Wait until app is restarted. 
See again the popup, there is an update available.

Even switching to run in background, gives same result. 

When checking in Settings - About for version, still see old version 4.8.0

**Comment by sdnts** [comment]
@calebrepkes Thanks for reporting this. 
To help us debug this, could you send us the zipped contents of the folder `~/Library/Application Support/com.postmanlabs.mac.ShipIt` to help@getpostman.com?

There should be three files there (`ShipIt_stderr.log`, `ShipIt_stdout.log` and `ShipItState.plist`)

**Comment by vegetableman** [comment]
@calebrepkes Could you also verify the permissions of the log files in the directory `~/Library/Application Support/com.postmanlabs.mac.ShipIt` as to whether you are the owner?

**Comment by vegetableman** [comment]
@calebrepkes Since you are facing a permission issue, for now, you could change the ownership of the above logs files to your user account, which would fix the update process, until we release a patch for it. Steps for the same:- http://osxdaily.com/2013/04/23/change-file-ownership-mac-os-x/

**Comment by jorbascrumps** [comment]
I'm also experiencing this issue and have sent my logs your way. Please let me know if I can help in any other ways.

**Comment by vegetableman** [comment]
@iamchristopher @calebrepkes  The fix will be part of the upcoming release `4.9.2`. Stay tuned ^_^.

**Comment by jorbascrumps** [comment]
Hi @vegetableman, I just received an update notification for `v4.9.2-canary.1` but the problem persists. Are you referring to another upcoming release?

**Comment by vegetableman** [comment]
@iamchristopher The problem is fixed on `v4.9.2-canary.1`. Any updates, henceforth, on the canary app should install and restart without any issues. You could then let me know if the problem still persists. Also, this issue will remain open, until the next two releases on the prod app.

**Comment by vegetableman** [comment]
@iamchristopher Also, thanks for testing the canary app 👍  ^_^.

**Comment by jorbascrumps** [comment]
Hi @vegetableman, thanks for looking into this. Unfortunately, I'm still experiencing this issue with `v4.9.2-canary.1` :(

**Comment by miguelzakharia** [comment]
I'm also experiencing this problem on the Mac client. I'm on version `4.10.2` and am seeing the message to upgrade to `4.10.3`. When I click "Restart" the app closes, but never restarts. When I open it again I'm prompted to download and install `4.10.3` once more.

I don't have `~/Library/Application Support/com.postmanlabs.mac.ShipIt` either so please let me know if I can send any other files for inspection.

**Comment by jorbascrumps** [comment]
Still seeing this issue, except my install has somehow reverted to `4.8.0-canary.1`?

**Comment by morleytatro** [comment]
Same issue here. I'm on 4.10.5.

**Comment by SamvelRaja** [comment]
@morleytatro can you provide the log from `~/Library/Application Support/com.postmanlabs.mac.ShipIt` to help[at]getpostman.com ?

**Comment by morleytatro** [comment]
I don't have that log (`com.postmanlabs.mac.ShipIt`) in `~/Library/Application Support` at all. There's a Postman folder inside it, but it's not there either. Where else might the log be?

**Comment by yunti** [comment]
I have the update issue too. (4.9.3) trying to update to version 5.0.0.  
What's the best way to update given the bug (whilst maintaining collections, history etc...)?

**Comment by pjamgithub** [comment]
Having the same issue with the 5.0.0 update.  I'm on 4.10.7 though

**Comment by daimor** [comment]
Update from 5.0.0 to 5.0.2 still not working. Shows `Restarts now to update`, but only stops and after start it manually, nothing was changed. And don't have `~/Library/Application Support/com.postmanlabs.mac.ShipIt`

**Comment by trigunam** [comment]
I have the exact same problem and no logs found in `~/Library/Application Support/com.postmanlabs.mac.ShipIt`

**Comment by phrazzld** [comment]
I'm trying to update from 4.10.7 to 5.0.2 but I get an Update Error: "Something went wrong. Please check your connection and try restarting the app."

No `~/Library/Application Support/com.postmanlabs.mac.ShipIt` file either.

**Comment by SamvelRaja** [comment]
Hey all,
The below process will help us in dig further on whats happening on background

**Opening up the app through terminal, so that we can get the logs on it.**
a. Right Click on the app -> Show packaged contents
b. The  go inside the folder ->  Contents -> MacOS 
c. Right Click on the Postman Executable and open with terminal
d. This will open the app for the user with console mode to track background data flow.

**Open up the devtools to check why the update dialogue is appearing.**
a. In the postman app devtools Cmd/Ctrl + Shift +i 
b. Goto -> console tab in devtools and type pm.app.getVersion() -> It should be the latest version `5.0.2`
c. Now Click the settings -> Check for updates options in postman app.
d. See the network tab in devtools
e. In the network tab there should be an API call that went in to check the updates 
- send us the Request and response section of it. 
- It should have returned Status code. **204 No Content**

f. If it is not returning 204, then the app tries to update, the background update log will be available in terminal opened in the Step 1. Send us this log.

**Comment by daimor** [comment]
Hey, I've managed to get my Postman updated automatically. After few attempts, I found this folder `/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt`. Where I found file `ShipIt_stderr.log` with lots of errors like this
```
2016-10-03 10:04:57.141 ShipIt[64842:2756627] Installation error: Error Domain=SQRLShipItRequestErrorDomain Code=2 "Could not read update request" UserInfo={NSLocalizedDescription=Could not read update request, NSUnderlyingError=0x7fce97602d70 {Error Domain=NSCocoaErrorDomain Code=260 "The file “ShipItState.plist” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/var/root/Library/Caches/com.postmanlabs.mac.ShipIt/ShipItState.plist, NSUnderlyingError=0x7fce97403780 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}}}
2016-10-03 10:04:57.142 ShipIt[64842:2756627] ShipIt quitting
```
Yes, I see that all errors dated so long ago. And also found some update folder, with Postman.app inside. Then, I decided to clean this folder. But looks like, update process, already worked, and in few seconds later Postman which was closed few time ago, unexpectedly opened. Usualy, when I click install and relaunch button, it s just closed without opening. And new stderr file appeared with content:
```
2017-07-07 13:46:37.776 ShipIt[33433:1154042] Beginning installation
2017-07-07 13:46:44.882 ShipIt[33433:1154046] Installation error: Error Domain=SQRLInstallerErrorDomain Code=-1 "Failed to copy bundle file:///Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/ to directory file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.1d592rO0/Postman.app" UserInfo={NSSourceFilePathErrorKey=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/Contents/Resources/app/node_modules/postman-runtime/test/integration/sanity/clear-vars-sandbox.test.js, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.1d592rO0/Postman.app/Contents/Resources/app/node_modules/postman-runtime/test/integration/sanity/clear-vars-sandbox.test.js, NSFilePath=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/Contents/Resources/app/node_modules/postman-runtime/test/integration/sanity/clear-vars-sandbox.test.js, NSUnderlyingError=0x7fac422974e0 {Error Domain=NSCocoaErrorDomain Code=4 "The file “clear-vars-sandbox.test.js” doesn’t exist." UserInfo={NSSourceFilePathErrorKey=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/Contents/Resources/app/node_modules/postman-runtime/test/integration/sanity/clear-vars-sandbox.test.js, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.1d592rO0/Postman.app/Contents/Resources/app/node_modules/postman-runtime/test/integration/sanity/clear-vars-sandbox.test.js, NSFilePath=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/Contents/Resources/app/node_modules/postman-runtime/test/integration/sanity/clear-vars-sandbox.test.js, NSUnderlyingError=0x7fac422a2a30 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}}, NSLocalizedDescription=Failed to copy bundle file:///Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/ to directory file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.1d592rO0/Postman.app}
2017-07-07 13:46:44.882 ShipIt[33433:1154046] ShipIt quitting
2017-07-07 13:46:44.950 ShipIt[33438:1154113] Resuming installation attempt 2
2017-07-07 13:46:44.965 ShipIt[33438:1154115] Installation error: Error Domain=SQRLInstallerErrorDomain Code=-1 "Failed to copy bundle file:///Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/ to directory file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.jEj4vani/Postman.app" UserInfo={NSSourceFilePathErrorKey=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/Contents/Resources/app/node_modules/postman-sandbox/node_modules/lodash/fp/lte.js, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.jEj4vani/Postman.app/Contents/Resources/app/node_modules/postman-sandbox/node_modules/lodash/fp/lte.js, NSFilePath=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/Contents/Resources/app/node_modules/postman-sandbox/node_modules/lodash/fp/lte.js, NSUnderlyingError=0x7f8603c1d0c0 {Error Domain=NSCocoaErrorDomain Code=4 "The file “lte.js” doesn’t exist." UserInfo={NSSourceFilePathErrorKey=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/Contents/Resources/app/node_modules/postman-sandbox/node_modules/lodash/fp/lte.js, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.jEj4vani/Postman.app/Contents/Resources/app/node_modules/postman-sandbox/node_modules/lodash/fp/lte.js, NSFilePath=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/Contents/Resources/app/node_modules/postman-sandbox/node_modules/lodash/fp/lte.js, NSUnderlyingError=0x7f8603c18d50 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}}, NSLocalizedDescription=Failed to copy bundle file:///Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/ to directory file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.jEj4vani/Postman.app}
2017-07-07 13:46:44.966 ShipIt[33438:1154115] ShipIt quitting
2017-07-07 13:46:47.018 ShipIt[33443:1154544] Resuming installation attempt 3
2017-07-07 13:46:47.037 ShipIt[33443:1154545] Installation error: Error Domain=SQRLInstallerErrorDomain Code=-1 "Failed to copy bundle file:///Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/ to directory file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.9Q6GRYlY/Postman.app" UserInfo={NSFilePath=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app, NSLocalizedDescription=Failed to copy bundle file:///Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app/ to directory file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.9Q6GRYlY/Postman.app, NSUnderlyingError=0x7fb956435f20 {Error Domain=NSCocoaErrorDomain Code=260 "The file “Postman.app” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/Users/DAiMor/Library/Caches/com.postmanlabs.mac.ShipIt/update.BNWhEW5/Postman.app, NSUnderlyingError=0x7fb956435db0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}}}
2017-07-07 13:46:47.038 ShipIt[33443:1154545] ShipIt quitting
2017-07-07 13:46:49.086 ShipIt[33444:1154571] Too many attempts to install, aborting update
2017-07-07 13:46:49.221 ShipIt[33444:1154573] Application launched at file:///Applications/Postman.app
2017-07-07 13:46:49.222 ShipIt[33444:1154573] ShipIt quitting
```
But it expected that update failed due to deleted update folder.
I clean folder again, and tried again
```
2017-07-07 13:49:52.544 ShipIt[33539:1157638] Beginning installation
2017-07-07 13:50:09.596 ShipIt[33539:1157406] Moved bundle from file:///Applications/Postman.app to file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.vkx7AzJ6/Postman.app
2017-07-07 13:50:09.596 ShipIt[33539:1157406] Moved bundle from file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.8zJzpS1v/Postman.app to file:///Applications/Postman.app
2017-07-07 13:50:11.810 ShipIt[33539:1157406] Couldn't remove owned bundle at location file:///var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.8zJzpS1v/Postman.app, error NSError { domain: NSCocoaErrorDomain, code: 4, description: "“Postman.app” couldn’t be removed.", failureReason: "The file doesn’t exist.", underlying error: NSError { domain: NSPOSIXErrorDomain, code: 2, description: "The operation couldn’t be completed. No such file or directory" }, userInfo: {
    NSFilePath = "/var/folders/wl/21g_gxfj4jlg9325829x34fc0000gp/T/com.postmanlabs.mac.ShipIt.8zJzpS1v/Postman.app";
    NSUserStringVariant =     (
        Remove
    );
} }
2017-07-07 13:50:14.857 ShipIt[33539:1157406] Installation completed successfully
2017-07-07 13:50:14.910 ShipIt[33539:1157835] Application launched at file:///Applications/Postman.app
2017-07-07 13:50:14.911 ShipIt[33539:1157835] ShipIt quitting
```
So, will see what happens with the next release.

**Comment by miguelzakharia** [comment]
The steps @daimor listed worked for me. I only needed to clean up the folder one time and Postman restarted as expected after update. A new error log was created though.

```
2017-07-07 11:40:06.676 ShipIt[50214:5424130] Beginning installation
2017-07-07 11:40:11.611 ShipIt[50214:5424543] Moved bundle from file:///Applications/Postman.app to file:///var/folders/4c/5jl1qn513yv8z4ydz9wc8dk00000gn/T/com.postmanlabs.mac.ShipIt.dxBskjIa/Postman.app
2017-07-07 11:40:11.612 ShipIt[50214:5424543] Moved bundle from file:///var/folders/4c/5jl1qn513yv8z4ydz9wc8dk00000gn/T/com.postmanlabs.mac.ShipIt.VgWkBqrR/Postman.app to file:///Applications/Postman.app
2017-07-07 11:40:12.642 ShipIt[50214:5424543] Couldn't remove owned bundle at location file:///var/folders/4c/5jl1qn513yv8z4ydz9wc8dk00000gn/T/com.postmanlabs.mac.ShipIt.VgWkBqrR/Postman.app, error NSError { domain: NSCocoaErrorDomain, code: 4, description: "“Postman.app” couldn’t be removed.", failureReason: "The file doesn’t exist.", underlying error: NSError { domain: NSPOSIXErrorDomain, code: 2, description: "The operation couldn’t be completed. No such file or directory" }, userInfo: {
    NSFilePath = "/var/folders/4c/5jl1qn513yv8z4ydz9wc8dk00000gn/T/com.postmanlabs.mac.ShipIt.VgWkBqrR/Postman.app";
    NSUserStringVariant =     (
        Remove
    );
} }
2017-07-07 11:40:14.249 ShipIt[50214:5424543] Installation completed successfully
2017-07-07 11:40:14.279 ShipIt[50214:5424697] Application launched at file:///Applications/Postman.app
2017-07-07 11:40:14.279 ShipIt[50214:5424697] ShipIt quitting
```

**Comment by morleytatro** [comment]
I'm not sure what changed on my end, but after following the first few steps as detailed by @SamvelRaja, I was able to successfully update the program.

**Comment by calle** [comment]
I'm also having trouble with automatic updates and have the following in my `~/Library/Caches/com.postmanlabs.mac.ShipIt/ShipIt_stderr.log`:
```
2017-08-31 07:52:26.859 ShipIt[3357:21747] Beginning installation
2017-08-31 07:52:26.956 ShipIt[3357:21748] Couldn't abort install and restore owned bundle to previous location file:///Applications/Postman.app, error NSError { domain: NSPOSIXErrorDomain, code: 2, description: "No such file or directory" }
2017-08-31 07:52:51.428 ShipIt[3357:22903] Installation error: Error Domain=NSPOSIXErrorDomain Code=13 "Permission denied" UserInfo={NSLocalizedDescription=Permission denied}
2017-08-31 07:52:51.428 ShipIt[3357:22903] ShipIt quitting
```

The reason seems to be that I try to update as a non-admin user (the `/Applications/Postman.app` is owned by the non-admin user) but the `/Applications` directory is owned by root. Similar to this issue https://github.com/Squirrel/Squirrel.Mac/pull/207

**Comment by sdnts** [comment]
@calle Indeed, based on our investigation, if you put the app in a directory that has the wrong permissions, the app cannot update, since the update will try to write to a directory that the application resides in.
A solution to this would be to move the app to a directory that has the correct permissions (like the `/Applications` directory on Mac).

**Comment by Mark90** [comment]
I had the issue of the update not even downloading, with the error message being "Please retry later". This happened on 5.0.0 when I wanted to update it to 5.1.3, and also on 5.1.3 when wanting to update to 5.2.0.

Just now, I started 5.1.3 through the terminal -as described by @SamvelRaja- in order to get logs, but for some reason this caused it to update without problems!

```
➜  ~ /Users/mark/Downloads/Postman-2.app/Contents/MacOS/Postman ; exit;
osx other channel release server url https://dl.pstmn.io/update/status?currentVersion=5.1.3&platform=osx&channel=stable
checking-for-update: {"sender":{"_events":{},"_eventsCount":5}}
update-available: {"sender":{"_events":{},"_eventsCount":5}}
2017-09-19 14:08:36.744 Postman[17132:351920] Download completed to: file:///Users/mark/Library/Caches/com.postmanlabs.mac.ShipIt/update.FeAzoo3/osx_64
update-downloaded: 
** Improvements **
- All new Status bar for quick access to useful actions
- Add ability to set a custom proxy to be used for requests made through the app
- Share collections and invite users to your team from the Team Library
- Introducing the PM API for cleaner and more robust pre-request & test scripts

** Bugfixes **
- Fixed an issue which prevented request reordering within a collection [GitHub#3291]
- Fixed various issues around Sync conflicts
- Fixed an issue that caused path variable values to be deleted when importing a collection
- Fixed an issue where Postman dynamic variables were not being populated in the autocomplete dropdown [GitHub#3354]
 Postman 5.2.0 https://dl.pstmn.io/download/version/5.2.0/osx_64
Closed Window (id: 1)

[Process completed]```

**Comment by ertyseidohl** [comment]
I fixed this issue by moving the Postman executable from my Downloads folder into my system-level Applications Folder.

**Comment by yunti** [comment]
@ertyseidohl Thanks , that worked perfectly for me.

**Comment by jemshit** [comment]
It is already in Applications folder, still update does not work. I click restart, Postman is closed, then never gets open again unless i open it myself. (and it didn't update). I have to manually download new postman from website and replace. (macOs sierra, postman 5.3.0)

**Comment by Mark90** [comment]
@jemshit Try opening the app from the terminal as described by @SamvelRaja up in this issue. This may "magically" resolve the issue, if not then you'll have logs to post here for debugging.

**Comment by jemshit** [comment]
@Mark90 I followed `e. In the network tab there should be an API call that went in to check the updates`, which showed me `update is already downloaded` insteading of sending request. So i clicked Update and Restart, which got stuck again. Current v 5.3.0. Log:

```
/Applications/Postman.app/Contents/MacOS/Postman ; e
xit;
osx other channel release server url https://dl.pstmn.io/update/status?currentVersion=5.3.0&platform=osx&channel=stable
checking-for-update: {"sender":{"_events":{},"_eventsCount":5}}
update-available: {"sender":{"_events":{},"_eventsCount":5}}
2017-10-23 12:11:44.184 Postman[17448:523733] Download completed to: file:///Users/jemshit/Library/Caches/com.postmanlabs.mac.ShipIt/update.RdzJ2p8/osx_64
update-downloaded: 
** Features **
- You can now add multiple user accounts and switch between them seamlessly

** Bugfixes **
- Fixed an issue where some OAuth 2.0 tokens were not being used with requests [GitHub#3608]
- Fixed an issue where getting a new OAuth 2.0 token was not working for some providers [GitHub#3604]
- Fixed an issue where getting a new OAuth 2.0 token flow would not complete [GitHub#3616
 Postman 5.3.1 https://dl.pstmn.io/download/version/5.3.1/osx_64
Closed Window (id: 1)
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
Deleting expired sessions...7 completed.

[Process completed]
```

**Comment by shabrad** [comment]
I have the same problem but with 5.4.1 installed and an update to 5.5.0 available.  With an update on Mac OS Sierra I normally have to supply a password for the sys admin for the mac but this upgrade doesn't ask for this.  I just have to click 'Remind me later' every time I run Postman.

**Comment by Adraesh** [comment]
Hi!

Any news on this topic?

![screen shot 2018-01-12 at 12 46 52 pm](https://user-images.githubusercontent.com/3873988/34871537-c43120f4-f796-11e7-8322-8c7b9f0b102a.png)
![screen shot 2018-01-12 at 12 46 42 pm](https://user-images.githubusercontent.com/3873988/34871538-c4515d10-f796-11e7-859a-420d4cc30689.png)
![screen shot 2018-01-12 at 12 46 36 pm](https://user-images.githubusercontent.com/3873988/34871539-c472345e-f796-11e7-8253-801189ec8c5c.png)

**Comment by sivcan** [comment]
@Adraesh - By any chance is your **Postman.app** inside your Downloads folder ?
If yes, move it to the **Applications** folder and then try updating.

**Comment by Adraesh** [comment]
@sivcan Thanks for your fast answer! My Postman.app is in the Application folder.

**Comment by sivcan** [comment]
@Adraesh - Did you try this https://github.com/postmanlabs/postman-app-support/issues/2489#issuecomment-313596858 as mentioned ? 
If yes, share the logs.

**Comment by Adraesh** [comment]
@sivcan Thanks it worked by starting the app with the console!

![screen shot 2018-01-12 at 2 06 04 pm](https://user-images.githubusercontent.com/3873988/34874491-ca5427aa-f7a1-11e7-9b87-139bc07e2f0a.png)

**Comment by sivcan** [comment]
@Adraesh - Awesome!

**Comment by MartinP7r** [comment]
Opening the app through console actually didn't help in my case. the file gets downloaded but then can't be installed due to permission error. Even when executed as root.  
```bash
2018-01-18 09:23:07.653 Postman[51622:1742541] Download completed to: file:///var/root/Library/Caches/com.postmanlabs.mac.ShipIt/update.ogV9xCo/osx_64
autoUpdate error: {} "ditto: /var/root/Library/Caches/com.postmanlabs.mac.ShipIt/update.ogV9xCo/osx_64: Permission denied"
```

**Comment by JRPeq** [comment]
sudo chown -R myUser /Applications/Postman.app/
chmod -R +x /Applications/Postman.app/ 

Solved my problem!

**Comment by marco-brandizi** [comment]
I think this is not fixed yet. The updater closes PM and then nothing else happens. I don't think changing system dir permissions or running PM as admin are acceptable solutions, for the standard way to manage this in MacOS is that the updater asks the user to provide the credentials of an admin user and then it runs stuff as that user. Putting the new .app file under user's Applications directory isn't very good either, since I want it available to the whole system.

**Comment by lucapalano** [comment]
Hi everyone, 

I also tried to run the Postman app from the terminal with following result:
> osx other channel release server url https://dl.pstmn.io/update/status?currentVersion=5.3.2&platform=osx&channel=stable
> checking-for-update: {"sender":{"_events":{},"_eventsCount":5}}
> update-available: {"sender":{"_events":{},"_eventsCount":5}}
> 2018-02-20 17:56:15.260 Postman[9155:927661] Download completed to: file:///Users/luca/Library/Caches/com.postmanlabs.mac.ShipIt/update.v7OPsE8/osx_64
> update-downloaded: 
> ** Bugfixes **
> - Fixed a rare issue where duplicating a folder added multiple copies of requests within the folder
>  Postman 5.5.3 https://dl.pstmn.io/download/version/5.5.3/osx_64
> Closed Window (id: 1)

No error was reported during the download process but, after restarting the Postman.app, the update download window appeared again.

I also attach the system console logs related to Postman.  Hope this will help.

[postman-console.txt](https://github.com/postmanlabs/postman-app-support/files/1741100/postman-console.txt)

Luca

**Comment by tusharsharma793** [comment]
Is their a way to disable the update download window prompt. I am using PM 5.5.0

**Comment by jennaprice** [comment]
I have the same issue. I am running 5.5.2 from my Applications folder and it asks to install a helper tool and I give it permission and it tries to update and it just downloads, closes and never opens. I will see if some of these things work.

**Comment by nicklaunches** [comment]
Same issue version 6.0.8. Tried from terminal didn't help

**Comment by naoko** [comment]
Same here with latest version v6.1.3 on Mac. Anyway to force-sync/download from cloud?

**Comment by Moes81** [comment]
Yepp, also here. But I'm still running v6.0.9

**Comment by luvianwang** [comment]
Same issue with latest version on Mac

**Comment by leblancexplores** [comment]
Postman is the new Firefox :(

**Comment by wouter-veeken** [comment]
For me the problem turned out to be **folder permissions**. I had Postman.app installed/running in my `Downloads` folder for some reason. Moved to `Applications`, and update works fine.

If your Postman.app is already in `Applications`, I'd recommend @JRPeq's solution posted above:
```
sudo chown -R {username} /Applications/Postman.app/
chmod -R +x /Applications/Postman.app/
```

**Comment by rberge** [comment]
![screen shot 2018-07-26 at 10 54 06 am](https://user-images.githubusercontent.com/20893688/43279378-a1c54bd8-90c2-11e8-9900-cc4735f41e3f.png)
![screen shot 2018-07-26 at 10 53 48 am](https://user-images.githubusercontent.com/20893688/43279387-a77a4c90-90c2-11e8-8a40-ab0a6dc5f012.png)

I am unable to upgrade Postman.  I have followed the @JRpeq solution (see attached).  For Postman I am in an endless loop where it asks to upgrade, I click to go ahead, after the download completes it usually asks if I want to restart and upgrade (not always sure because it seems flakey).  When I press to restart and upgrade nothing happens.

**Comment by asmittal** [comment]
Even with latest version, upgrade doesn't work. :( 
Pity a small issue is not resolved even after 2 years.

**Comment by Quezion** [comment]
I've been trying to update Postman also for around 2 years without success. The `chown` solution didn't work & moving it to the user's `Applications` also failed.

Stuck in the same bootloop that @rberge wrote up.

I'll try a manual uninstall + reinstall later, but this permission issue broke 2 years of Postman software updates for me and _still_ isn't fixed. Not a good look.

**Comment by Quezion** [comment]
I tried the solution posted by @daimor from a while back with no luck.

What got me updated for today is launching Postman with root and updating there (no signin, just skip to app), then closing & manually relaunching as root.

After closing the second root instance I could launch the updated app on my normal account.

With the disclaimer that running as root is a bad idea for many security reasons, you can use the below snippet to do this (assuming Postman is in your root /Applications.)

```
sudo /Applications/Postman.app/Contents/MacOS/Postman
```

I'll note that I manage the Postman accounts & billing for my team, and this experience has soured me on software that has otherwise been solid.

If an automated software solution could not be deployed via Postman, the developers should've recapped this thread with "please run this workaround as a 1-time deal to get your Postman working again."

As it is now, I assume I'll need to manually repeat this process for every Postman update until I retire this laptop.

**Comment by skyvondave** [comment]
Clicking "Update Postman, Restart Now" from the wrench icon with the orange dot just closes Postman. Nothing else. Upon restart, it's the same version 6.6.1 with the same update notification.

![image](https://user-images.githubusercontent.com/40671781/52676267-d4db2c00-2ee6-11e9-97f1-af507adc87ee.png)

**Comment by danieljanderson** [comment]
I am surprised that this is still an issue.  I tried moving it to my users app folder.  That didnt work. I tried @wouter-veeken  solution and that didnt work.

**Comment by lucapalano** [comment]
With last postman releases, this issue looks like to be fixed.

**Comment by asmittal** [comment]
Currently on 7.5   . Says 7.6 available > Asked for update and restart > Accepted < Application closed but didn't start. 
Manually started the app, and it is still on 7.5

**Comment by kamalaknn** [comment]
@asmittal Can you share the application logs from `View -> Developer -> View logs in Finder` to us by sending them over email to `help@getpostman.com`? Please provide the GitHub issue id in the mail.

**Comment by ash007-postman** [comment]
@kamalaknn I have sent you the requested logs from @asmittal

**Comment by kamalaknn** [comment]
Hi @danieljanderson @asmittal @lucapalano 

Can you give the details of the operating system you are using?

1. Type of operating system (macOS/windows)
2. Version (10.14/10.15)

**Comment by asmittal** [comment]
MAC 10.14.6

Regards,
Amit

**Comment by a85** [comment]
Closing this issue as we haven't seen widespread reports around it. Specific installation issues might be due to the environment or permissions. Please open a new issue with replication steps if you see a problem.

**Comment by skyvondave** [comment]
63 comments isn't widespread? We've had problems with it off and on, and some of our clients use it. If it's a problem, leave it open until it's fixed; don't close it because it's not bothersome enough.

**Comment by marco-brandizi** [comment]
@wouter-veeken, as I've already wrote long ago, @JRPeq's solution isn't acceptable for a serious user: I don't want to lower my system's security by changing an app's permissions or ownership just because its upgrader doesn't work properly. The proper way to upgrade in macOS is asking the non-admin user to type the credentials of another user which has admin rights, then running the installation procedure with that, and having the latter setting the right permissions for the installed files (eg, `chmod -R u=rwX,go=rX ...`).

Currently, the PM upgrade procedure doesn't work with regular users (ie, what any self-respected hacker should normally use) precisely because it doesn't do that switch to admin, and the solution is making it work that way, not me setting less safe permissions/ownership.

Until this is fixed (and I understand that maybe it will never happen), I'll have to do everything manually as if I was at first install: trash the old .app, download the new one and place it where the old was (eg, `/Applications`, but I usually prefer to have `/Applications/Local/SomethingElse/Sample.app`). Not a big deal, but doing it the proper way would be easier and I don't get why PM developers have left this simple issue behind for almost three years, only to eventually decide to close it because "there aren't widespread reports".

**Comment by breathermachine** [comment]
I just want to add that Postman doesn't let you update if it doesn't reside in either `~/Applications` or `/Applications`

**Comment by manmeet90** [comment]
This is what worked for me on MacOs 11.x.x
1. Close the Postman app if open.
2. make sure your postman app is in `/Applications` folder.If not first move the installed Postman app to /Applications
3. Open the `terminal` app.
4. start Postman from the terminal by running the command `Postman.app/Contents/MacOS/Postman` 

 This should launch the postman app. Now try to download the update You should be able to download and install the update package.

**Comment by marco-brandizi** [comment]
@manmeet90, this usually doesn't work unless you're an admin user when running the app from the terminal. Trying to first become admin (`su`, or `sudo`) doesn't work either, cause only the current user can open graphical applications on its own desktop.

Bottom line: it seems developers don't want to provide a proper way to upgrade their app under MacOS (they closed this bug two years ago, without a fix). For this and similar cases, I found a fairly good solution by using [HomeBrew](https://brew.sh/): I've installed PostMan via HomeBrew and I use it to upgrade. It's not like updates managed from within the app (when they work), but it's fairly smooth.
